### PR TITLE
Remove Rwlock of KV client.

### PR DESCRIPTION
**Pull request by pzheng1025**
Use insert and delete sequence to make sure consistency.

### CODE REVIEW

File: etcd-client/src/kv/mod.rs [hunk 1]
```diff
@@ -310,23 +314,22 @@ impl Kv {
...
-                        if let Err(e) = inner.watch_sender.send(watch_request).await {
-                            warn!(
-                                "Fail to send watch request, the error is {}, clean cache",
-                                e
-                            );
-                            drop(cache_lock);
-                            inner.clean_cache().await;
-                            break;
-                        }
+        if inner.cache_enable && inner.cache_availability.load(Ordering::Acquire) {
```
Comment (pzheng1025) [inline etcd-client/src/kv/mod.rs]: May miss watch event.

File: etcd-client/src/kv/mod.rs [hunk 2]
```diff
@@ -109,7 +109,7 @@ impl KvInner {
             cache,
             cache_enable,
             watch_sender: watch_req_sender,
-            cache_availability: RwLock::new(()),
+            cache_availability: AtomicBool::from(true),
```
Comment (mrwwheat7) [inline etcd-client/src/kv/mod.rs:112]: Any reason we change to AtomicBool here?

File: etcd-client/src/kv/cache.rs [hunk 3]
```diff
@@ -118,11 +124,12 @@ impl Cache {
     /// Clean cache
     #[allow(unsafe_code)]
     pub async fn clean(&self) {
+        // Don't change clear sequence
```
Comment (mrwwheat7) [inline etcd-client/src/kv/cache.rs:123]: Why do we change the sequence here?

File: etcd-client/src/kv/mod.rs (hunk 2 above)
Comment (pzheng1025) [reply etcd-client/src/kv/mod.rs:112]: RwLock is a little heavy than AtomicBool. We use function call sequence to make sure consistency.

File: etcd-client/src/kv/cache.rs (hunk 3 above)
Comment (pzheng1025) [reply etcd-client/src/kv/cache.rs:123]: This clear sequence will make sure consistency between hashtable and lru-queue.

**Review by mrwwheat7 — COMMENTED** [review]
LGTM

File: etcd-client/src/kv/mod.rs [hunk 4]
```diff
@@ -175,31 +175,45 @@ impl KvInner {
...
+                                                    .insert(key.clone(), event.get_kv().clone())
                                                     .await;
+                                                cache_inner.insert_watch_id(key.clone(), watch_id);
                                             }
                                         }
                                     } else {
-                                        cache_inner
-                                            .delete(event.get_kv().get_key().to_vec(), true)
-                                            .await;
+                                        cache_inner.delete(event.get_kv().get_key().to_vec()).await;
```
Comment (pzheng1025) [inline etcd-client/src/kv/mod.rs:213]: check version.

**Pull request closed without merging**